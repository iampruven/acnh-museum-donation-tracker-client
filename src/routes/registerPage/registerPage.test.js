import React from 'react';
import ReactDOM from 'react-dom';
import RegisterPage from './registerPage';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
//smoke test
describe('renders without crashing',()=>{
    it('it renders without crashing',()=>{
        const div = document.createElement('div');

        ReactDOM.render(<MemoryRouter><RegisterPage /></MemoryRouter>,div);
        ReactDOM.unmountComponentAtNode(div);
    })


//snapshot test
    it('renders the UI as expected', ()=>{
        const tree = renderer
            .create(<MemoryRouter><RegisterPage /></MemoryRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })



})