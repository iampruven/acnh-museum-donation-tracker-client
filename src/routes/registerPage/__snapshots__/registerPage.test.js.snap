// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders the UI as expected 1`] = `
<div>
  <section
    className="backdrop"
  >
    <h1
      className="centered"
    >
      Register
    </h1>
    <form
      className="column centered"
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        id="username"
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <span>
        Password must contain a min of 8 characters and must contain at least one upper case, lower case, number and special character.
      </span>
      <input
        id="password"
        name="password"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="btn"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      className="alert"
    />
  </section>
</div>
`;
